flytekit/types/structured/structured_dataset.py

```diff
@@ -0,0 +1,380 @@
+        _datetime.timedelta: type_models.LiteralType(simple=type_models.SimpleType.DURATION),
+        _np.string_: type_models.LiteralType(simple=type_models.SimpleType.STRING),
+        _np.str_: type_models.LiteralType(simple=type_models.SimpleType.STRING),
+        _np.object_: type_models.LiteralType(simple=type_models.SimpleType.STRING),
+        str: type_models.LiteralType(simple=type_models.SimpleType.STRING),
+    }
+
+    DATASET_DECODING_HANDLERS: Dict[Type[Any], Dict[Type[Any], DatasetDecodingHandler]] = {}
+    DATASET_ENCODING_HANDLERS: Dict[Type[Any], Dict[Type[Any], DatasetEncodingHandler]] = {}
+    DATASET_PERSISTENCE_HANDLERS: Dict[Type[typing.Any], Dict[Type[Any], DatasetPersistenceHandler]] = {}
+    DATASET_RETRIEVAL_HANDLERS: Dict[Type[Any], Dict[Type[Any], DatasetRetrievalHandler]] = {}
+    Handlers = typing.Union[
+        DatasetDecodingHandler, DatasetEncodingHandler, DatasetPersistenceHandler, DatasetRetrievalHandler
+    ]
+    _REGISTER_TYPES: List[Type] = []
+
+    def __init__(self):
+        super().__init__("StructuredDataset Transformer", StructuredDataset)
+
+    def _get_dataset_column_literal_type(self, t: Type):
+        if t in self._SUPPORTED_TYPES:
+            return self._SUPPORTED_TYPES[t]
+        if hasattr(t, "__origin__") and t.__origin__ == list:
+            return type_models.LiteralType(collection_type=self._get_dataset_column_literal_type(t.__args__[0]))
+        if hasattr(t, "__origin__") and t.__origin__ == dict:
+            return type_models.LiteralType(map_value_type=self._get_dataset_column_literal_type(t.__args__[1]))
+        raise AssertionError(f"type {t} is currently not supported by StructuredDataset")
+
+    def _get_dataset_type(self, t: Type[StructuredDataset]) -> StructuredDatasetType:
+        # TODO: Convert pandas, arrow, and other dataframes column datatype to StructuredDatasetType
+        if t != StructuredDataset:
+            return StructuredDatasetType(columns=[])
+        converted_cols: typing.List[StructuredDatasetType.DatasetColumn] = []
+        for k, v in t.columns().items():
+            lt = self._get_dataset_column_literal_type(v)
+            converted_cols.append(StructuredDatasetType.DatasetColumn(name=k, literal_type=lt))
+        return StructuredDatasetType(columns=converted_cols)
+
+    def register_handler(
+        self,
+        from_type: Type[T],
+        to_type: Type[T],
+        h: Handlers,
+    ):
+        """
+        Call this with any handler to register it with this dataframe meta-transformer
+        """
+        if from_type not in self._REGISTER_TYPES:
+            self._REGISTER_TYPES.append(from_type)
+            TypeEngine.override_transformer(self, from_type)
+
+        if to_type not in self._REGISTER_TYPES:
+            self._REGISTER_TYPES.append(to_type)
+            TypeEngine.override_transformer(self, to_type)
+
+        registry: dict
+
+        if isinstance(h, DatasetRetrievalHandler):
```

Comment:
i think we should make these issubclass.  See the other comment about registering classes instead of instances of the class.